### PR TITLE
Added debug_traceBlock* methods

### PR DESCRIPTION
**Pull request by MadelineMurray**
Added debug_traceBlock* methods

### CODE REVIEW

File: docs/Reference/Pantheon-API-Methods.md [hunk 1]
```diff
@@ -2703,6 +2703,163 @@ Reruns the transaction with the same state as when the transaction was executed.
     }
     ```
 
+### debug_traceBlock
+
+Returns full stack trace of all invoked opcodes of all transactions included in the block.
```
Comment (ajsutton) [inline docs/Reference/Pantheon-API-Methods.md]:
```suggestion
Returns full trace of all invoked opcodes of all transactions included in the block.
```

While the trace can contain the stack after each opcode, it's not a stack trace as such.

File: docs/Reference/Pantheon-API-Methods.md [hunk 2]
```diff
@@ -2703,6 +2703,163 @@ Reruns the transaction with the same state as when the transaction was executed.
...
+          "memory" : [ ],
+          "storage" : null
+        } ]
+      }
+    }
+    ```    
+
+### debug_traceBlockByHash
+
+Returns full stack trace of all invoked opcodes of all transactions included in the block.
```
Comment (ajsutton) [inline docs/Reference/Pantheon-API-Methods.md]:
```suggestion
Returns full trace of all invoked opcodes of all transactions included in the block.
```

File: docs/Reference/Pantheon-API-Methods.md [hunk 3]
```diff
@@ -2703,6 +2703,163 @@ Reruns the transaction with the same state as when the transaction was executed.
...
+                    }
+                ]
+            }
+        ]
+    }
+    ```
+    
+### debug_traceBlockByNumber
+
+Returns full stack trace of all invoked opcodes of all transactions included in the block.
```
Comment (ajsutton) [inline docs/Reference/Pantheon-API-Methods.md]:
```suggestion
Returns full trace of all invoked opcodes of all transactions included in the block.
```

File: docs/Reference/Pantheon-API-Methods.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-API-Methods.md]: Fixed

File: docs/Reference/Pantheon-API-Methods.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-API-Methods.md]: Fixed

File: docs/Reference/Pantheon-API-Methods.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-API-Methods.md]: Fixed

**Pull request merged**